core/src/main/java/com/linecorp/armeria/internal/common/stream/EmptyFixedStreamMessage.java

```diff
@@ -16,19 +16,51 @@
 
 package com.linecorp.armeria.internal.common.stream;
 
+import static java.util.Objects.requireNonNull;
+
 import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.reactivestreams.Subscriber;
 
 import com.google.common.collect.ImmutableList;
 
 import com.linecorp.armeria.common.annotation.Nullable;
 import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.stream.SubscriptionOption;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+
+import io.netty.util.concurrent.EventExecutor;
 
 /**
  * A {@link FixedStreamMessage} that publishes no objects, just a close event.
  */
 @UnstableApi
 public class EmptyFixedStreamMessage<T> extends FixedStreamMessage<T> {
 
+    @Override
+    public boolean isComplete() {
+        return whenComplete().isDone();
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super T> subscriber, EventExecutor executor,
+                          SubscriptionOption... options) {
+        requireNonNull(subscriber, "subscriber");
+        requireNonNull(executor, "executor");
+        executor.execute(() -> {
```

Comment:
Do we need to reschedule this even when it is in the event loop?

Reply:
No it's not. let me fix that. 😉